test/SlowTests/Server/Documents/DataArchival/DataArchivalDataSubscriptionsTests.cs

```diff
@@ -0,0 +1,107 @@
+using Raven.Client.Util;
+using SlowTests.Core.Utils.Entities;
+using Sparrow;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Server.Documents.DataArchival;
+
+
+    public class DataArchivalDataSubscriptionsTests: RavenTestBase
+    {
+        public DataArchivalDataSubscriptionsTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private async Task SetupDataArchival(IDocumentStore store)
+        {
+            var config = new DataArchivalConfiguration {Disabled = false, ArchiveFrequencyInSec = 100};
+
+            await DataArchivalHelper.SetupDataArchival(store, Server.ServerStore, config);
+        }
+        
+        
+    
+        [Fact]
+        public async Task ArchivedDocumentsDataSubscription()
+        {
+            using (var store = GetDocumentStore())
+            {
+                List<Company> companies = new(); 
+                
+                // Insert document with archive time before activating the archival
+                var company = new Company {Name = "Company Name", Address1 = "Dabrowskiego 6"};
+                var retires = SystemTime.UtcNow.AddMinutes(5);
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(company);
+                    var metadata = session.Advanced.GetMetadataFor(company);
+                    metadata[Constants.Documents.Metadata.ArchiveAt] = retires.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                    await session.SaveChangesAsync();
+                }
+                
+                // Set-up the subscription and run the worker
+                var subsId = await store.Subscriptions.CreateAsync(new SubscriptionCreationOptions
+                {
+                    Query = "from Companies",
+                    Name = "Created",
+                    SourceItemKind = IndexSourceItemKind.ArchivedOnly
+                });
+                var worker = store.Subscriptions.GetSubscriptionWorker<Company>(subsId);
+                var t = worker.Run(batch => companies.AddRange(batch.Items.Select(item => item.Result)));
+
+                WaitForValue(() => companies.Count, 1, 5000);
+                Assert.Equal(companies.Count, 0);
+
+                // Activate the archival manually
+                await SetupDataArchival(store);
+                var database = await Databases.GetDocumentDatabaseInstanceFor(store);
+                database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+                var documentsArchiver = database.DataArchivist;
+                await documentsArchiver.ArchiveDocs();
+
+                await AssertWaitForCountAsync(() => Task.FromResult(companies), 1);
+                Assert.Equal(companies.Count, 1);
+            }
+        }
+        
+        [Fact]
+        public async Task ArchivedDocumentsDataSubscription_CanDisableAndEnableSubscription_SourceItemKindWillBePersisted()
```

Comment:
What is this test going to test exactly? It says about "persisted"? What persistence is it about? Shouldn't we reload a database or something like that?

Reply:
94bf30efee5c41abff31d63dda4af0815ad53314 fixed here, it used to don't work earlier, so to be sure I've wrote a test